utils/test-utils/index.js

```diff
@@ -6,6 +6,19 @@ const renderTwigFile = (file, options, cb) =>
     cb(err, format(html, { parser: 'html' }))
   );
 
+const renderTwigFileAsNode = (file, options) =>
+  new Promise((resolve, reject) =>
+    Twig.renderFile(file, options, (err, html) => {
+      if (err) return reject(err);
+
+      const div = document.createElement('div');
+      div.innerHTML = html.trim();
+
+      return resolve(div.firstChild);
+    })
+  );
+
 module.exports = {
   renderTwigFile,
+  renderTwigFileAsNode,
```

Comment:
for now, we expose the 2 methods but at some point, we should probably harmonize our tests and only use one

unless we want to test the ouput of the twig template as well. but then, should we still run it through prettier?

Reply:
My opinion: `renderTwigFileAsNode` achieves what is necessary.
For what it's worth as feedback from the runner, here's what I get when a difference appears:

```
 FAIL  src/ec/examples/composition/composition.test.js
  EC - Composition
    ✕ renders correctly (57ms)

  ● EC - Composition › renders correctly

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot "EC - Composition renders correctly 1".

    - Snapshot
    + Received

    @@ -3,11 +3,11 @@
          ><span class="ecl-button__label">Example button</span></span
        >
      </button>
      
      <blockquote class="ecl-blockquote">
    -   <p class="ecl-blockquote__body">Example body</p>
    +   <span class="ecl-blockquote__body">Example body</span>
        <footer class="ecl-blockquote__attribution">
          <cite class="ecl-blockquote__author">Author name</cite>
        </footer>
      </blockquote>
      "
```

It's as good as it could be, nothing is lost with comparison to the other method. Snapshots are also more readable.

Reply:
and as expected, when you change the order of CSS classes (which doesn't affect the end result):

```diff
          type="submit"
        >
          <span class="ecl-button__container"
            ><span class="ecl-button__label">Search</span
            ><svg
-             class="ecl-icon ecl-icon--fluid ecl-button__icon ecl-button__icon--after"
+             class="ecl-icon ecl-icon--fluid ecl-button__icon--after ecl-button__icon"
              focusable="false"
              aria-hidden="true"
            >
              <use xlink:href="static/icons.svg#general--search"></use></svg
          ></span>
```

😢

Reply:
Yes, that is the case with the `renderTwigFile` as well. Yesterday, when I was adding checks on which classes to add and which not, the order was caught again as an error from jest. It's not a problem, I think, as long as the person knows the `u` in `watchAll` ...